source/patterns/@aws-solutions-constructs/core/lib/cloudfront-distribution-helper.ts

```diff
@@ -183,6 +204,17 @@ export function CloudFrontOriginAccessIdentity(scope: Construct, comment?: strin
   });
 }
 
+function validateSecurityHeadersBehavior(
```

Comment:
Our pattern is to put checks like these much earlier in the process - preferably in `defaults.CheckProps(props);`

This flags the error much sooner, plus if there are multiple issues with the client's input we find them all at once - rather than the annoying pattern of the client fixing one, discovering another, fixing that, discovering another, etc.

Reply:
Okay. I'm not quite happy to validate httpSecurityHeaders and responseHeadersPolicyProps before httpSecurityHeaders is assigned a default (line 139). My validation has to treat 'undefined' like 'true' then (because 'undefined' will lead to the default 'true' being used) - that's counterintuitive, 'undefined' is usually treated like 'false' in JavaScript.
But I agree with the goal of finding all issues at once.
I'll move the validation to `defaults.CheckProps(props);`